### ISSUE TITLE
Rule Change: (object-curly-newline) Separate ObjectPattern and ObjectDestructuring or some other way do differentiate

### ISSUE THREAD

**Issue by crazyjat** [eslint/eslint#17802]
### What rule do you want to change?

object-curly-newline

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
const { foo, bar, foobar } = someOtherVariable;
```

### What does the rule currently do for this code?

Using the existing rule, `ObjectPattern` will also apply to object destructuring. Due to the fact that there is currently no eslint rule available to enforce object property on new line for destructured object, it would be very nice to be able to disable the newline behavior for this case.

### What will the rule do after it's changed?

```
        "object-curly-newline": [
            "error",
            {
                "ObjectExpression": {
                    "minProperties": 2,
                    "multiline": true
                },
                "ObjectPattern": {
                    "minProperties": 2,
                    "multiline": true
                },
                "ObjectDestructuring": {
                    "multiline": false
                },
                "ExportDeclaration": {
                    "minProperties": 2,
                    "multiline": true
                },
                "ImportDeclaration": {
                    "minProperties": 2,
                    "multiline": true
                }
            }
        ],
```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

I have my rules setup to ensure that any time an array or object is defined, if there are multiple properties, each property should appear on its own line. This works for objects, array, imports, exports, but not for object destructuring. The `eslint/object-property-newline` doesn't work for destructuring, nor does `@stylistic/object-property-newline`, nor have I been able to find any existing rule that does this.

**Comment by Tanujkanti4441** [comment]
Hi @crazyjat thanks for the issue, but since `object-curly-newline` rule is deprecated now so not looking for new changes there.
